src/parsing/parseTakerBidEvent.ts

```diff
@@ -83,52 +89,92 @@ export const parseTakerBidEvent: ParserMethod<TakerBid> = async ({
+      hash: bid.transactionHash as `0x${string}`,
+    });
 
-      const parsedLogs = parseEventLogs({
-        abi: HypercertMinterAbi,
-        logs: transactionLogs,
-      });
+    // parse logs to get claimID, contractAddress and cid
+    const transactionLogsHypercertMinter = transactionReceipt.logs.filter(
+      (log) =>
+        log.address.toLowerCase() ===
+        addresses?.HypercertMinterUUPS?.toLowerCase(),
+    );
 
-      // Look for both BatchValueTransfer and TransferSingle events
-      const batchValueTransferLog = parsedLogs.find(
-        // @ts-expect-error eventName is missing in the type
-        (log) => log.eventName === "BatchValueTransfer"
+    const parsedLogs = parseEventLogs({
+      abi: HypercertMinterAbi,
+      logs: transactionLogsHypercertMinter,
+    });
+
+    // Look for both BatchValueTransfer and TransferSingle events
+    const batchValueTransferLog = parsedLogs.find(
+      // @ts-expect-error eventName is missing in the type
+      (log) => log.eventName === "BatchValueTransfer",
+    );
+    const transferSingleLog = parsedLogs.find(
+      // @ts-expect-error eventName is missing in the type
+      (log) => log.eventName === "TransferSingle",
+    );
+
+    // Get the claim ID from either event type
+    let claimId;
+    // @ts-expect-error args is missing in the type
+    if (batchValueTransferLog?.args?.claimIDs?.[0]) {
+      // @ts-expect-error args is missing in the type
+      claimId = batchValueTransferLog.args.claimIDs[0];
+    // @ts-expect-error args is missing in the type
+    } else if (transferSingleLog?.args?.id) {
+      // In this case, the ID from the transferSingleLog is a fraction token ID
+      // We need to get the claim ID from the fraction token ID
+      // @ts-expect-error args is missing in the type
+      claimId = getHypercertTokenId(transferSingleLog.args.id);
+    }
+
+    if (!claimId) {
+      throw new Error(
+        "Failed to find claim ID in BatchValueTransfer or TransferSingle events",
       );
-      const transferSingleLog = parsedLogs.find(
-        // @ts-expect-error eventName is missing in the type
-        (log) => log.eventName === "TransferSingle"
+    }
+
+    const hypercertId = `${chain_id}-${getAddress(bid.params?.collection)}-${claimId}`;
+
+    let currencyAmount = 0n;
+    const currency = getAddress(bid.params.currency);
+    if (currency === zeroAddress) {
+      // Get value of the transaction
+      const transaction = await client.getTransaction({
+        hash: bid.transactionHash as `0x${string}`,
+      });
+      currencyAmount = transaction.value;
+    } else {
+      const currencyLogs = transactionReceipt.logs.filter(
+        (log) => log.address.toLowerCase() === currency.toLowerCase(),
       );
+      const parsedCurrencyLogs = parseEventLogs({
+        abi: erc20Abi,
+        logs: currencyLogs,
+      });
+      const transferLog = parsedCurrencyLogs.find(
```

Comment:
Same as in the script, there'll be mutliple ERC20 Transfer logs

Reply:
Updated